### ISSUE TITLE
Jakub: Training & deployment tasks for hackathon deadline

### ISSUE THREAD

**Issue by lbartoszcze** [wisent-ai/OpenEnv#2]
Hi Jakub! Here's a summary of what's needed on your side to get the paper and demo across the finish line. The game implementations, evaluation pipeline, and paper structure are all in place — we're blocked on trained model checkpoints to populate results.

## What's ready

- **110+ games** fully implemented and tested (744 tests passing)
- **Paper restructured** to 14 sections + appendices covering the full scope (commit `9c756d0`)
- **Tournament runner** ready to evaluate any checkpoint across all games
- **Safety benchmark pipeline** ready (HarmBench, ETHICS, TruthfulQA, XSTest, MT-Bench) — just needs model paths
- **Evaluation metrics** implemented (cooperation rate, exploitation resistance, Pareto efficiency, fairness, adaptability, strategic reasoning)

### 1. GRPO training run (P0)

`train/train.py` is set up with TRL. Please run a full training on the H100 and push the checkpoint. Key things:

- Use the full game library (not just the base 6 games) if possible
- Save the checkpoint somewhere we can both access (HF Hub or shared storage)
- Save training logs/curves — we'll need them for the paper

### 2. DPO training run (P0)

After GRPO produces trajectories, collect preference pairs (cooperative vs defecting trajectories) and run DPO. We need both Kant-GRPO and Kant-DPO checkpoints to compare in the paper.

### 3. Space deployment validation (P0)

- `spaces/kant/README.md` still has the echo-environment placeholder text — please update it to describe KantBench
- Verify the Space serves the full game library, not just base games
- Test the WebSocket flow end-to-end

### 4. Training documentation (P1)

Once training is done, please note down:
- Final hyperparameters (learning rate, batch size, group size for GRPO, beta for DPO)
- Curriculum phases if any (which games in which order)
- Compute used (GPU hours, hardware)
- Reward shaping weights

I'll put these into the paper's appendix.

## Dependency chain

```
Jakub: GRPO training → checkpoint
Jakub: DPO training → checkpoint
    ↓
Lukasz: Tournament evaluation → metrics
Lukasz: Safety benchmarks → transfer results
    ↓
Both: Populate paper results → final compile
```

The full task list is in #1. Let me know if you need anything from my side or if the training setup needs changes. Thanks!
